### DISCUSSION TITLE
Add hydrogens with correct protonation state

### DISCUSSION THREAD

**Discussion by hl2500** [rdkit/rdkit#5860] [Q&A] [answered]
Hello, I'm wondering if rdkit could consider the correct protonation state of a molecule when adding hydrogens. For example, some nitrogens have +1 charge (e.g., -NH3(+), -NH2-(+)) under physiological PH, but AddHs() function only protonate those nitrogen to the neutral form (e.g., -NH2, -NH-). Thanks.

**Comment by greglandrum** [comment, marked as answer]
@hl2500 : the RDKit adds the number of Hs which is appropriate for the charge state you provide.

We do not currently have a model which figures out what the charge/protonation state should be at physiological pH. That is, unfortunately, information that you need to provide yourself.

**Comment by gmseabra** [comment]
Hi,

Since thie thread is already ~4y old, I was wondering if there has been any change to this issue. Does RDKIT now offer any way to calculate protonation states for molecules or protein residues?

Thanks!

**Reply by greglandrum** [reply]
@gmseabra unfortunately not. That would require a high-quality open-source model for assigning protonation states with a license and implementation style that we could integrate into the RDKit. I am not aware of any such thing, but I would be happy to learn otherwise.

**Reply by gmseabra** [reply]
Thanks. How about [Dimorphite_DL](https://github.com/durrantlab/dimorphite_dl)? Do you think it could be useful to integrate on RDKit?

**Reply by greglandrum** [reply]
> Thanks. How about [Dimorphite_DL](https://github.com/durrantlab/dimorphite_dl)? Do you think it could be useful to integrate on RDKit?

I don't think I see a benefit to doing that? I think it's pretty easy to install the standalone Dimorphite package together with the RDKit (in fact, Dimorphite uses the RDKit itself). We don't need to have everything integrated into the core RDKit package.

**Reply by gmseabra** [reply]
Fair enough :-)

My point was more on the existence of a model to assign protonation states.

One question,  though: can one at least claim that the protonation states chosen by RDKit conform to a _"standard state at neutral pH"_? Or are the hydrogens added just to complete valencies?

Thanks!

**Reply by greglandrum** [reply]
> 
> My point was more on the existence of a model to assign protonation states.

Fair. :-)


> One question, though: can one at least claim that the protonation states chosen by RDKit conform to a _"standard state at neutral pH"_? Or are the hydrogens added just to complete valencies?

Hs are added solely to complete valences based on the charges in the input structure.
